lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -543,6 +543,10 @@ public function getOrCreateDocuments($className, $nodes, array &$hints = array()
                 $documentState[$class->parentMapping] = $this->getOrCreateProxyFromNode($node->getParent(), $locale);
             }
 
+            if ($class->depthMapping) {
+                $documentState[$class->depthMapping] = $node->getDepth();
```

Comment:
Where is the best place to test that this actually works without writing an entire new testcase?